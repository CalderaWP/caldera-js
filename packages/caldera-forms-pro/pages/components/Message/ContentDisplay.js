import React from 'react';

export const ContentDisplay = ({content}) => {
	const createMarkup = () => {
		const {rendered} = content;
		return {__html:rendered};
	};
	return <div dangerouslySetInnerHTML={createMarkup()}/>


};

