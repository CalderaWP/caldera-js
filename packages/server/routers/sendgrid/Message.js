const EmailAddress = require( './EmailAddress');
const Personalizations = require( './Personalizations');
module.exports = function Message({
	to,
	subject,
	fromName,
	contentType = 'text/html',
	attatchments,
	message,
	replyTo,
	cc=null,
	bcc = null
				 }) {

	contentType = 'text/html'  === contentType || 'text/plain' === contentType
		? contentType
		: 'text/html';
	const personalizations = Personalizations({to, subject,cc,bcc});
	replyTo = new EmailAddress(replyTo);
	return {
		method: 'POST',
		path: '/v3/mail/send',
		body: {
			reply_to: replyTo,
			personalizations,
			from: {email: 'no-reply@calderaformspro.net', name: fromName},
			content: [
				{
					type: contentType,
					value: message
				},
			],
		},
	}
};
