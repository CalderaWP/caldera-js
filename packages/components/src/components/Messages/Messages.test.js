import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import { Message, MESSAGE_CLASS } from './Message';
import { messageObjectFactory } from './messageObjectFactory';

describe('Message component', () => {
	it('Shows a message', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Good Things Happened'
				}}
			/>
		);
		expect(wrapper).toHaveLength(1);
	});

	it('Renders message', () => {
		const wrapper = renderer.create(
			<Message
				message={{
					message: 'Hi Roy',
					error: false
				}}
			/>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('Renders error message', () => {
		const wrapper = renderer.create(
			<Message
				message={{
					message: 'An error happened',
					error: true
				}}
			/>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('Renders no error message', () => {
		const wrapper = renderer.create(
			<Message
				message={{
					error: true
				}}
			/>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('Shows the right message message', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Hi Roy'
				}}
			/>
		);
		expect(wrapper.text()).toBe('Hi Roy');
	});

	it('shows classname prop as class', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Hi Roy'
				}}
				className={'mike'}
			/>
		);
		expect(wrapper.find('.mike')).toHaveLength(1);
	});

	it('Has error class when is an error', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Something bad has happened.',
					error: true
				}}
			/>
		);
		expect(wrapper.find('.has-error')).toHaveLength(1);
	});

	it('Does not have error class when is not an error', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Something good has happened.',
					error: false
				}}
			/>
		);
		expect(wrapper.find('.has-error')).toHaveLength(0);
	});

	it('Does have caldera-components-error class when an error', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Something',
					error: true
				}}
			/>
		);
		expect(wrapper.hasClass(MESSAGE_CLASS)).toEqual(true);
	});

	it('Does have caldera-components-error class when not an error', () => {
		const wrapper = shallow(
			<Message
				message={{
					message: 'Something',
					error: false
				}}
			/>
		);
		expect(wrapper.hasClass(MESSAGE_CLASS)).toEqual(true);
	});
});

describe('messageObjectFactory', () => {
	it('casts truthy error to boolean', () => {
		expect(
			messageObjectFactory({
				error: 1,
				message: 'Roy'
			}).error
		).toBe(true);
	});

	it('casts falsey error to boolean', () => {
		expect(
			messageObjectFactory({
				error: 'false',
				message: 'Roy'
			}).error
		).toBe(false);
	});

	it('Adds missing message key', () => {
		expect(
			messageObjectFactory({
				error: 'false'
			}).message
		).toBe('');
	});
});
