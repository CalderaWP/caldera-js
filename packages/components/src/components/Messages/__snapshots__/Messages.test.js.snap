// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Renders error message 1`] = `
<div
  className="caldera-components-message has-error caldera-components-error"
>
  An error happened
</div>
`;

exports[`Message component Renders message 1`] = `
<div
  className="caldera-components-message caldera-components-not-error"
>
  Hi Roy
</div>
`;

exports[`Message component Renders no error message 1`] = `null`;
